### PR TITLE
Exception renaming

### PR DESCRIPTION
**Pull request by OH72**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/main/java/com/softserve/teachua/dto/club/validation/DescriptionValidation.java [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.softserve.teachua.exception.BadRequestException;
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/dto/club/validation/DescriptionValidation.java:7]: unnecessary "import"

File: src/main/java/com/softserve/teachua/controller/ClubController.java [hunk 2]
```diff
@@ -5,7 +5,6 @@
 import com.softserve.teachua.dto.search.AdvancedSearchClubProfile;
 import com.softserve.teachua.dto.search.SearchClubProfile;
 import com.softserve.teachua.dto.search.SimilarClubProfile;
-import com.softserve.teachua.exception.BadRequestException;
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/controller/ClubController.java:8]: unnecessary "import"

File: src/main/java/com/softserve/teachua/dto/club/validation/PhoneValidation.java [hunk 3]
```diff
@@ -20,13 +20,13 @@
...
+                throw new IncorrectInputException("Phone number length must be 10");
             }
             if (phoneNumber.matches(".*[a-zA-Z]+.*")) {
-                throw new BadRequestException("Phone number cannot contain letters.");
+                throw new IncorrectInputException("Phone number cannot contain letters.");
             }
             if(phoneNumber.matches(".*[$&+,:;=?@#|'<>.-^*()%!]+.*")) {
-                throw new BadRequestException("Phone number contains illegal symbols");
+                throw new IncorrectInputException("Phone number contains illegal symbols");
             }
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/dto/club/validation/PhoneValidation.java:30]: It is correct to use for validation IllegalArgumentException (IncorrectInputException extends IllegalArgumentException)

File: src/main/java/com/softserve/teachua/exception/BadRequestException.java [hunk 4]
```diff
@@ -11,7 +12,7 @@
  *
  * @author Denis Burko
  */
-public class BadRequestException extends IllegalArgumentException {
+public class BadRequestException extends IllegalStateException {
     public static final String CANT_FIND_FIELD_IN_JSON = "Can't find %s in JSON body";
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/exception/BadRequestException.java:16]: BadRequest shouldnt extends IllegalArgumentException

File: src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java [hunk 5]
```diff
@@ -84,10 +84,10 @@ public void deleteFile(String filePath) {
             return;
         }
         if (filePath == null || filePath.isEmpty()) {
-            throw new BadRequestException("File path can not be null or empty");
+            throw new IncorrectInputException("File path can not be null or empty");
         }
         if (!filePath.contains(UPLOAD_LOCATION)) {
-            throw new BadRequestException("Wrong uploaded file path");
+            throw new IncorrectInputException("Wrong uploaded file path");
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/FileUploadServiceImpl.java:90]: It is correctly to use IllegalArgumentException if fields have wrong values (IncorrectInputException extends IllegalArgumentException)

**Review by VictoriaAmazonka — APPROVED** [review]
Thank you!

File: src/main/java/com/softserve/teachua/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -204,7 +203,7 @@ public SuccessRegistration registerUser(UserProfile userProfile) {
         }
 
         if ("ROLE_ADMIN".equals(userProfile.getRoleName())) {
-            throw new BadRequestException("Illegal role argument: ROLE_ADMIN");
+            throw new IncorrectInputException("Illegal role argument: ROLE_ADMIN");
```
Comment (OH72) [inline src/main/java/com/softserve/teachua/service/impl/UserServiceImpl.java:206]: It is correctly to use IllegalArgumentException if fields have wrong values (IncorrectInputException extends IllegalArgumentException)

**Pull request merged**